### ISSUE TITLE
CRLF is treated as two newlines when skip_empty_rows = FALSE

### ISSUE THREAD

**Issue by nacnudus** [tidyverse/readr#972]
Windows newlines `\r\n` are treated as two new lines when `skip_empty_rows = FALSE`.

``` r
library(readr)

# This is the output I would expect.
read_csv("foo\n\nbar", col_names = FALSE, skip_empty_rows = FALSE)
#> # A tibble: 3 x 1
#>   X1   
#>   <chr>
#> 1 foo  
#> 2 <NA> 
#> 3 bar

# I would expect the output to be the same as above.
read_csv("foo\r\n\r\nbar", col_names = FALSE, skip_empty_rows = FALSE)
#> # A tibble: 4 x 1
#>   X1   
#>   <chr>
#> 1 foo  
#> 2 <NA> 
#> 3 <NA> 
#> 4 bar
```

Created on 2019-02-28 by the [reprex package](https://reprex.tidyverse.org) (v0.2.0.9000).

<details>

<summary>Session info</summary>

``` r
devtools::session_info()
#> ─ Session info ──────────────────────────────────────────────────────────
#>  setting  value                       
#>  version  R version 3.5.2 (2018-12-20)
#>  os       Arch Linux                  
#>  system   x86_64, linux-gnu           
#>  ui       X11                         
#>  language                             
#>  collate  en_NZ.UTF-8                 
#>  ctype    en_GB.UTF-8                 
#>  tz       Europe/London               
#>  date     2019-02-28                  
#> 
#> ─ Packages ──────────────────────────────────────────────────────────────
#>  package     * version    date       lib source                           
#>  assertthat    0.2.0      2017-04-11 [1] CRAN (R 3.5.0)                   
#>  backports     1.1.3      2018-12-14 [1] CRAN (R 3.5.2)                   
#>  callr         3.1.1      2018-12-21 [1] CRAN (R 3.5.2)                   
#>  cli           1.0.1      2018-09-25 [1] CRAN (R 3.5.1)                   
#>  crayon        1.3.4      2017-09-16 [1] CRAN (R 3.5.0)                   
#>  desc          1.2.0      2018-05-01 [1] CRAN (R 3.5.0)                   
#>  devtools      2.0.1.9000 2019-01-28 [1] Github (r-lib/devtools@e4e57aa)  
#>  digest        0.6.18     2018-10-10 [1] CRAN (R 3.5.1)                   
#>  evaluate      0.12       2018-10-09 [1] CRAN (R 3.5.1)                   
#>  fansi         0.4.0      2018-11-09 [1] Github (brodieG/fansi@ab11e9c)   
#>  fs            1.2.6      2018-08-23 [1] CRAN (R 3.5.2)                   
#>  glue          1.3.0.9000 2019-01-28 [1] Github (tidyverse/glue@8188cea)  
#>  highr         0.7        2018-06-09 [1] CRAN (R 3.5.1)                   
#>  hms           0.4.2.9001 2019-02-28 [1] Github (tidyverse/hms@16ff76e)   
#>  htmltools     0.3.6      2017-04-28 [1] CRAN (R 3.5.0)                   
#>  knitr         1.21       2018-12-10 [1] CRAN (R 3.5.1)                   
#>  magrittr      1.5        2014-11-22 [1] CRAN (R 3.5.0)                   
#>  memoise       1.1.0      2017-04-21 [1] CRAN (R 3.5.0)                   
#>  nvimcom     * 0.9-75     2019-01-03 [1] local                            
#>  pillar        1.3.1.9000 2019-01-23 [1] Github (r-lib/pillar@3a54b8d)    
#>  pkgbuild      1.0.2      2018-10-16 [1] CRAN (R 3.5.1)                   
#>  pkgconfig     2.0.2      2018-08-16 [1] CRAN (R 3.5.1)                   
#>  pkgload       1.0.2      2018-10-29 [1] CRAN (R 3.5.1)                   
#>  prettyunits   1.0.2      2015-07-13 [1] CRAN (R 3.5.0)                   
#>  processx      3.2.1      2018-12-05 [1] CRAN (R 3.5.1)                   
#>  ps            1.3.0      2018-12-21 [1] CRAN (R 3.5.2)                   
#>  R6            2.4.0      2019-02-14 [1] CRAN (R 3.5.2)                   
#>  Rcpp          1.0.0      2018-11-07 [1] CRAN (R 3.5.2)                   
#>  readr       * 1.3.1.9000 2019-02-28 [1] Github (tidyverse/readr@b7e0b99) 
#>  remotes       2.0.2      2018-10-30 [1] CRAN (R 3.5.2)                   
#>  rlang         0.3.1      2019-01-08 [1] CRAN (R 3.5.2)                   
#>  rmarkdown     1.11       2018-12-08 [1] CRAN (R 3.5.1)                   
#>  rprojroot     1.3-2      2018-01-03 [1] CRAN (R 3.5.0)                   
#>  sessioninfo   1.1.1      2018-11-05 [1] CRAN (R 3.5.1)                   
#>  stringi       1.3.1      2019-02-13 [1] CRAN (R 3.5.2)                   
#>  stringr       1.4.0      2019-02-10 [1] CRAN (R 3.5.2)                   
#>  testthat      2.0.1      2018-10-13 [1] CRAN (R 3.5.2)                   
#>  tibble        2.0.1.9001 2019-02-28 [1] Github (tidyverse/tibble@92f5604)
#>  usethis       1.4.0      2018-08-14 [1] CRAN (R 3.5.1)                   
#>  utf8          1.1.4      2018-05-24 [1] CRAN (R 3.5.0)                   
#>  withr         2.1.2      2018-03-15 [1] CRAN (R 3.5.0)                   
#>  xfun          0.4        2018-10-23 [1] CRAN (R 3.5.1)                   
#>  yaml          2.2.0      2018-07-25 [1] CRAN (R 3.5.1)                   
#> 
#> [1] /home/nacnudus/R/x86_64-pc-linux-gnu-library/3.5
#> [2] /usr/lib/R/library
```

</details>

**Comment by vpanfilov** [comment]
I also have problems reading from file with CRLFs on Windows regardless of `skip_empty_rows ` value:

``` r
txt <- "1\r\n2\r\n3\r\n4\r\n"
readr::read_lines(txt)
#> [1] "1" "2" "3" "4"

readr::read_lines(txt, skip_empty_rows = TRUE)
#> [1] "1" "2" "3" "4"

file <- tempfile()

writeLines(txt, con = file)

readr::read_lines(file)
#> [1] "1" ""  "2" ""  "3" ""  "4" ""  ""

readr::read_lines(file, skip_empty_rows = TRUE)
#> [1] "1"   "\n2" "\n3" "\n4" "\n"
```

<sup>Created on 2019-03-04 by the [reprex package](https://reprex.tidyverse.org) (v0.2.1)</sup>
<details><summary>Session info</summary>

``` r
devtools::session_info()
#> - Session info ----------------------------------------------------------
#>  setting  value                       
#>  version  R version 3.5.2 (2018-12-20)
#>  os       Windows 10 x64              
#>  system   x86_64, mingw32             
#>  ui       RTerm                       
#>  language (EN)                        
#>  collate  Russian_Russia.1251         
#>  ctype    Russian_Russia.1251         
#>  tz       Europe/Moscow               
#>  date     2019-03-04                  
#> 
#> - Packages --------------------------------------------------------------
#>  package     * version date       lib source        
#>  assertthat    0.2.0   2017-04-11 [1] CRAN (R 3.5.0)
#>  backports     1.1.3   2018-12-14 [1] CRAN (R 3.5.1)
#>  callr         3.1.1   2018-12-21 [1] CRAN (R 3.5.2)
#>  cli           1.0.1   2018-09-25 [1] CRAN (R 3.5.1)
#>  crayon        1.3.4   2017-09-16 [1] CRAN (R 3.5.0)
#>  desc          1.2.0   2018-05-01 [1] CRAN (R 3.5.1)
#>  devtools      2.0.1   2018-10-26 [1] CRAN (R 3.5.1)
#>  digest        0.6.18  2018-10-10 [1] CRAN (R 3.5.1)
#>  evaluate      0.13    2019-02-12 [1] CRAN (R 3.5.2)
#>  fs            1.2.6   2018-08-23 [1] CRAN (R 3.5.1)
#>  glue          1.3.0   2018-07-17 [1] CRAN (R 3.5.1)
#>  highr         0.7     2018-06-09 [1] CRAN (R 3.5.0)
#>  hms           0.4.2   2018-03-10 [1] CRAN (R 3.5.0)
#>  htmltools     0.3.6   2017-04-28 [1] CRAN (R 3.5.0)
#>  knitr         1.21    2018-12-10 [1] CRAN (R 3.5.1)
#>  magrittr      1.5     2014-11-22 [1] CRAN (R 3.5.0)
#>  memoise       1.1.0   2017-04-21 [1] CRAN (R 3.5.0)
#>  pillar        1.3.1   2018-12-15 [1] CRAN (R 3.5.1)
#>  pkgbuild      1.0.2   2018-10-16 [1] CRAN (R 3.5.1)
#>  pkgconfig     2.0.2   2018-08-16 [1] CRAN (R 3.5.1)
#>  pkgload       1.0.2   2018-10-29 [1] CRAN (R 3.5.1)
#>  prettyunits   1.0.2   2015-07-13 [1] CRAN (R 3.5.0)
#>  processx      3.2.1   2018-12-05 [1] CRAN (R 3.5.1)
#>  ps            1.3.0   2018-12-21 [1] CRAN (R 3.5.2)
#>  R6            2.4.0   2019-02-14 [1] CRAN (R 3.5.2)
#>  Rcpp          1.0.0   2018-11-07 [1] CRAN (R 3.5.1)
#>  readr         1.3.1   2018-12-21 [1] CRAN (R 3.5.2)
#>  remotes       2.0.2   2018-10-30 [1] CRAN (R 3.5.1)
#>  rlang         0.3.1   2019-01-08 [1] CRAN (R 3.5.2)
#>  rmarkdown     1.11    2018-12-08 [1] CRAN (R 3.5.1)
#>  rprojroot     1.3-2   2018-01-03 [1] CRAN (R 3.5.0)
#>  sessioninfo   1.1.1   2018-11-05 [1] CRAN (R 3.5.1)
#>  stringi       1.3.1   2019-02-13 [1] CRAN (R 3.5.2)
#>  stringr       1.4.0   2019-02-10 [1] CRAN (R 3.5.2)
#>  testthat      2.0.1   2018-10-13 [1] CRAN (R 3.5.1)
#>  tibble        2.0.1   2019-01-12 [1] CRAN (R 3.5.2)
#>  usethis       1.4.0   2018-08-14 [1] CRAN (R 3.5.1)
#>  withr         2.1.2   2018-03-15 [1] CRAN (R 3.5.0)
#>  xfun          0.5     2019-02-20 [1] CRAN (R 3.5.2)
#>  yaml          2.2.0   2018-07-25 [1] CRAN (R 3.5.0)
#> 
#> [1] C:/Users/ishutov/Documents/R/win-library/3.5
#> [2] C:/Program Files/R/R-3.5.2/library
```

</details>

**Comment by netique** [comment]
I've observed a simillar and presumably unintended behavior using `read_lines()` with `skip_empty_rows = TRUE`. When a empty line is preceded with non-empty,  line following that empty line is prepended with `\n`. At least on Windows.

```
read_lines("my_file", skip_empty_rows = TRUE)
#> [1] "\r\n2\r\n\r\n4\r\n"

read_lines("Makefile", skip_empty_rows = TRUE) 
#> [1] "2"   "\n4"
```

When I "roundtrip" the file, i.e. first reading it, then writing with `write_lines`, everything appears to be OK.

**Comment by jimhester** [comment]
Fixed by #1172
